test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs

```diff
@@ -2085,6 +2085,181 @@ public void PackCommand_NuspecFileWithTokens()
+
+                // Assert
+
+                // Verify the nuget pack command exit code
+                Assert.NotNull(commandRunner);
+                Assert.Equal(1, commandRunner.Item1);
+
+                var exepctedError = string.Format(
+                    "Unable to output resolved nuspec file because it would overwrite the original at '{0}\\{1}'\r\n",
+                    workingDirectory, 
+                    PackageName);
+
+                Assert.Equal(exepctedError, commandRunner.Item3);
+            }
+        }
+
+        [Fact]
+        public void PackCommand_InstallPackageToV3Feed()
+        {
+            using (var workingDirectory = TestDirectory.Create())
+            {
+                // Arrange
+                var packageName = "packageA.nuspec";
+                var originalDirectory = Path.Combine(workingDirectory, "Original");
+                Directory.CreateDirectory(originalDirectory);
+
+                Util.CreateFile(
+                    originalDirectory,
+                    packageName,
+@"<?xml version=""1.0"" encoding=""utf-8""?>
+<package xmlns='http://schemas.microsoft.com/packaging/2011/08/nuspec.xsd'>
+  <metadata minClientVersion=""3.3"">
+    <id>packageA</id>
+    <version>1.2.3.4</version>
+    <title>packageATitle</title>
+    <authors>$author$</authors>
+    <owners>testOwner</owners>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <description>the description</description>
+    <copyright>Copyright (C) Microsoft 2013</copyright>
+    <tags>Microsoft,Sample,$tagVar$</tags>
+    <frameworkAssemblies>
+      <frameworkAssembly assemblyName=""System"" />
+    </frameworkAssemblies>
+  </metadata>
+</package>");
+
+                // Act
+
+                // Execute the pack command and feed in some properties for token replacements and 
+                // set the flag to save the resolved nuspec to output directory.\
+                var arguments = string.Format(
+                    "pack {0} -properties tagVar=CustomTag;author=test1@microsoft.com -InstallPackageToV3Feed",
+                    Path.Combine(originalDirectory, packageName));
+
+                var commandRunner = CommandRunner.Run(
+                    Util.GetNuGetExePath(),
+                    workingDirectory,
+                    arguments,
+                    waitForExit: true);
+
+                // Assert
+
+                // Verify the nuget pack command exit code
+                Assert.NotNull(commandRunner);
+                Assert.True(
+                    0 == commandRunner.Item1,
+                    string.Format("{0} {1}", commandRunner.Item2 ?? "null", commandRunner.Item3 ?? "null"));
+
+                // Verify the zip file has the resolved nuspec
+                XDocument nuspecZipXml = null;
+                var nupkgPath = Path.Combine(workingDirectory, "packageA.1.2.3.4.nupkg");
+                using (var zip = new ZipArchive(File.OpenRead(nupkgPath)))
```

Comment:
we have PackageArchiveReader and NuspecReader to do these tasks.
https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs#L877-L880

Reply:
Resolved